crates/matrix-sdk-indexeddb/src/event_cache_store/mod.rs

```diff
@@ -0,0 +1,421 @@
+        Ok(None)
+    }
+
+    /// Remove a media file's content from the media store.
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - The `MediaRequest` of the file.
+    async fn remove_media_content(&self, request: &MediaRequestParameters) -> Result<()> {
+        Ok(())
+    }
+
+    /// Get a media file's content associated to an `MxcUri` from the
+    /// media store.
+    ///
+    /// In theory, there could be several files stored using the same URI and a
+    /// different `MediaFormat`. This API is meant to be used with a media file
+    /// that has only been stored with a single format.
+    ///
+    /// If there are several media files for a given URI in different formats,
+    /// this API will only return one of them. Which one is left as an
+    /// implementation detail.
+    ///
+    /// # Arguments
+    ///
+    /// * `uri` - The `MxcUri` of the media file.
+    async fn get_media_content_for_uri(&self, uri: &MxcUri) -> Result<Option<Vec<u8>>> {
+        Ok(None)
+    }
+
+    /// Remove all the media files' content associated to an `MxcUri` from the
+    /// media store.
+    ///
+    /// This should not raise an error when the `uri` parameter points to an
+    /// unknown media, and it should return an Ok result in this case.
+    ///
+    /// # Arguments
+    ///
+    /// * `uri` - The `MxcUri` of the media files.
+    async fn remove_media_content_for_uri(&self, uri: &MxcUri) -> Result<()> {
+        Ok(())
+    }
+
+    fn media_retention_policy(&self) -> MediaRetentionPolicy {
+        // TODO on the sqlite version this has a media_service... what is that?
+        // It seems there is a Trait EventCacheStoreMedia that might need to be implemented
+        MediaRetentionPolicy::default()
+    }
+
+    /// Set the `MediaRetentionPolicy` to use for deciding whether to store or
+    /// keep media content.
+    ///
+    /// # Arguments
+    ///
+    /// * `policy` - The `MediaRetentionPolicy` to use.
+    async fn set_media_retention_policy(&self, policy: MediaRetentionPolicy) -> Result<()> {
+        Ok(())
+    }
+
+    /// Set whether the current [`MediaRetentionPolicy`] should be ignored for
+    /// the media.
+    ///
+    /// The change will be taken into account in the next cleanup.
+    ///
+    /// # Arguments
+    ///
+    /// * `request` - The `MediaRequestParameters` of the file.
+    ///
+    /// * `ignore_policy` - Whether the current `MediaRetentionPolicy` should be
+    ///   ignored.
+    async fn set_ignore_media_retention_policy(
+        &self,
+        request: &MediaRequestParameters,
+        ignore_policy: IgnoreMediaRetentionPolicy,
+    ) -> Result<()> {
+        Ok(())
+    }
+
+    /// Clean up the media cache with the current `MediaRetentionPolicy`.
+    ///
+    /// If there is already an ongoing cleanup, this is a noop.
+    async fn clean_up_media_cache(&self) -> Result<()> {
+        Ok(())
+    }
+
+    async fn try_take_leased_lock(
```

Comment:
Copy and paste from crypto_store. I could not find any info on what this "leased_lock" is.

Reply:
This is related to [the cross-process lock](https://matrix-org.github.io/matrix-rust-sdk/matrix_sdk/store_locks/index.html), the doc is missing a high-level overview I think…

Reply:
I just copied the same from the crypto_store, if you think it is fine as it is, then no need. It should just work, right?

Reply:
Possible candidate to try to put into a shared mod as well.